addOns/spider/src/main/java/org/zaproxy/addon/spider/parser/SpiderGitParser.java

```diff
@@ -0,0 +1,372 @@
+                                    + indexEntryExtendedFlag);
+                }
+
+                // specific to version 3 and above, if the extended flag is set for the entry.
+                if (indexFileVersion > 2 && indexEntryExtendedFlag == 1) {
+                    if (getLogger().isDebugEnabled())
+                        getLogger()
+                                .debug(
+                                        "For Index file version "
+                                                + indexFileVersion
+                                                + ", reading an extra 16 bits for Entry "
+                                                + entryIndex);
+                    short indexEntryExtendedFlags = dataBuffer.getShort();
+                    entryBytesRead += 2;
+                    if (getLogger().isDebugEnabled())
+                        getLogger()
+                                .debug(
+                                        "Entry "
+                                                + entryIndex
+                                                + " has (optional) extended flags "
+                                                + indexEntryExtendedFlags);
+                }
+
+                String indexEntryName = null;
+                if (indexFileVersion > 3) {
+                    if (getLogger().isDebugEnabled())
+                        getLogger()
+                                .debug(
+                                        "Inflating the (deflated) entry name for index entry "
+                                                + entryIndex
+                                                + " based on the previous entry name, since Index file version "
+                                                + indexFileVersion
+                                                + " requires this");
+
+                    // get bytes until we find one with the msb NOT set. count the bytes.
+                    int n = 0, removeNfromPreviousName = 0;
+                    byte msbsetmask = (byte) (1 << 7); // 1000 0000
+                    byte msbunsetmask = (byte) ((~msbsetmask) & 0xFF); // 0111 1111
+                    while (++n > 0) {
+                        byte byteRead = dataBuffer.get();
+                        entryBytesRead++;
+                        if (n == 1) // zero the msb of the first byte read
+                        removeNfromPreviousName =
+                                    (removeNfromPreviousName << 8)
+                                            | (0xFF & (byteRead & msbunsetmask));
+                        else // set the msb of subsequent bytes read
+                        removeNfromPreviousName =
+                                    (removeNfromPreviousName << 8)
+                                            | (0xFF & (byteRead | msbsetmask));
+                        if ((byteRead & msbsetmask) == 0)
+                            break; // break if msb is NOT set in the byte
```

Comment:
LGTM points out that n controlling the loop is always greater than 0, which is true. I guess I could move the increment of n inside the loop and use a boolean for control instead of the break?
`boolean msbSet = true;` then set it to false instead of breaking.

Reply:
While keeping the `++n` in the loop?

Reply:
Yes

Reply:
Ok, looks good then. Could also be `while (true)`.